opslevel/resource_opslevel_integration_azure_resources.go

```diff
@@ -0,0 +1,223 @@
+}
+
+type IntegrationAzureResourcesResource struct {
+	CommonResourceClient
+}
+
+type IntegrationAzureResourcesResourceModel struct {
+	Aliases        types.List   `tfsdk:"aliases"`
+	ClientId       types.String `tfsdk:"client_id"`
+	ClientSecret   types.String `tfsdk:"client_secret"`
+	Id             types.String `tfsdk:"id"`
+	LastSyncedAt   types.String `tfsdk:"last_synced_at"`
+	Name           types.String `tfsdk:"name"`
+	SubscriptionId types.String `tfsdk:"subscription_id"`
+	TenantId       types.String `tfsdk:"tenant_id"`
+}
+
+func NewIntegrationAzureResourcesResourceModel(ctx context.Context, planModel IntegrationAzureResourcesResourceModel, azureResourcesIntegration opslevel.Integration) (IntegrationAzureResourcesResourceModel, diag.Diagnostics) {
+	resourceModel := IntegrationAzureResourcesResourceModel{
+		ClientId:       planModel.ClientId,
+		ClientSecret:   planModel.ClientSecret,
+		Id:             ComputedStringValue(string(azureResourcesIntegration.Id)),
+		Name:           RequiredStringValue(azureResourcesIntegration.Name),
+		SubscriptionId: RequiredStringValue(azureResourcesIntegration.SubscriptionId),
+		TenantId:       RequiredStringValue(azureResourcesIntegration.TenantId),
+	}
+
+	if azureResourcesIntegration.LastSyncedAt == nil {
+		resourceModel.LastSyncedAt = types.StringNull()
+	} else {
+		resourceModel.LastSyncedAt = types.StringValue(azureResourcesIntegration.LastSyncedAt.String())
+	}
+	var diags diag.Diagnostics
+	resourceModel.Aliases, diags = types.ListValueFrom(ctx, types.StringType, azureResourcesIntegration.Aliases)
+
+	return resourceModel, diags
+}
+
+func (r *IntegrationAzureResourcesResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_integration_azure_resources"
+}
+
+func (r *IntegrationAzureResourcesResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Azure Resources Integration resource",
+
+		Attributes: map[string]schema.Attribute{
+			"client_id": schema.StringAttribute{
+				Description: "The client OpsLevel uses to access the Azure account.",
+				Required:    true,
+			},
+			"client_secret": schema.StringAttribute{
+				Description: "The client secret OpsLevel uses to access the Azure account.",
+				Required:    true,
+				Sensitive:   true,
+			},
+			"tenant_id": schema.StringAttribute{
+				Description: "The tenant OpsLevel uses to access the Azure account.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"subscription_id": schema.StringAttribute{
+				Description: "The subscription OpsLevel uses to access the Azure account.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"last_synced_at": schema.StringAttribute{
```

Comment:
this isn't a field on the AWS integration - are we sure it'll be in the public API?

Reply:
@rocktavious this will be in the public schema and this is for the Azure Resources integration only.